aws_batch/pipeline.py

```diff
@@ -457,88 +492,106 @@ def run_sample(sample_s3_input_path, sample_s3_output_path,
-        result_dir + '/' + FILTER_DEUTEROSTOME_FROM_TAXID_ANNOTATED_FASTA_OUT,
-        result_dir + '/' + GENERATE_TAXID_ANNOTATED_FASTA_FROM_RAPSEARCH2_M8_OUT,
-        'NR',
-        result_dir, sample_s3_output_path, lazy_run=False)
- 
+                    result_dir + '/' + RAPSEARCH2_OUT,
+                    result_dir + '/' + FILTER_DEUTEROSTOME_FROM_TAXID_ANNOTATED_FASTA_OUT,
+                    result_dir + '/' + GENERATE_TAXID_ANNOTATED_FASTA_FROM_RAPSEARCH2_M8_OUT,
+                    'NR',
+                    result_dir, sample_s3_output_path, lazy_run=False)
+
+    run_filter_deuterostomes_from_m8(sample_name,
+                    result_dir + '/' + ANNOTATE_RAPSEARCH2_M8_WITH_TAXIDS_OUT,
+                    result_dir + '/' + FILTER_DEUTEROSTOMES_FROM_NR_M8_OUT,
+                    deuterostome_list_s3_path,
+                    result_dir, sample_s3_output_path, lazy_run=False)
+
     run_generate_taxid_outputs_from_m8(sample_name,
-        result_dir + '/' + ANNOTATE_RAPSEARCH2_M8_WITH_TAXIDS_OUT,
-        fastq_file_1,
-        result_dir + '/' + NR_M8_TO_TAXID_COUNTS_FILE_OUT,
-        result_dir + '/' + NR_TAXID_COUNTS_TO_JSON_OUT,
-        result_dir + '/' + NR_TAXID_COUNTS_TO_SPECIES_RPM_OUT,
-        result_dir + '/' + NR_TAXID_COUNTS_TO_GENUS_RPM_OUT,
-        taxid2info_s3_path, 'NR', db_sample_id, 
-        sample_host, sample_location, sample_date, sample_tissue,
-        sample_template, sample_library, sample_sequencer, sample_notes,
-        result_dir, sample_s3_output_path, lazy_run=False)
+                    result_dir + '/' + FILTER_DEUTEROSTOMES_FROM_NR_M8_OUT,
+                    fastq_file_1,
+                    result_dir + '/' + NR_M8_TO_TAXID_COUNTS_FILE_OUT,
+                    result_dir + '/' + NR_TAXID_COUNTS_TO_JSON_OUT,
+                    result_dir + '/' + NR_TAXID_COUNTS_TO_SPECIES_RPM_OUT,
+                    result_dir + '/' + NR_TAXID_COUNTS_TO_GENUS_RPM_OUT,
+                    taxid2info_s3_path, 'NR', db_sample_id,
+                    sample_host, sample_location, sample_date, sample_tissue,
+                    sample_template, sample_library, sample_sequencer, sample_notes,
+                    result_dir, sample_s3_output_path, lazy_run=False)
 
     run_combine_json_outputs(sample_name,
-        result_dir + '/' + NT_TAXID_COUNTS_TO_JSON_OUT,
-        result_dir + '/' + NR_TAXID_COUNTS_TO_JSON_OUT,
-        result_dir + '/' + COMBINED_JSON_OUT,
-        result_dir, sample_s3_output_path, lazy_run=False)
+                    result_dir + '/' + NT_TAXID_COUNTS_TO_JSON_OUT,
+                    result_dir + '/' + NR_TAXID_COUNTS_TO_JSON_OUT,
+                    result_dir + '/' + COMBINED_JSON_OUT,
+                    result_dir, sample_s3_output_path, lazy_run=False)
 
 def run_star(sample_name, fastq_file_1, fastq_file_2, star_genome_s3_path,
              result_dir, scratch_dir, sample_s3_output_path, lazy_run):
     logger = logging.getLogger()
```

Comment:
why did you strip out all the early return for the lazy run? It makes the code a lot harder to read. and it's the coding convention people follow a lot.  Also at some point we need to follow the 80-char length. This would make things really hard.